cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -159,24 +159,89 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
+            /*List<String> errorsInConfig = configuration.validate(versions.version(spec.getVersion()));
+            if (!errorsInConfig.isEmpty()) {
+                for (String error : errorsInConfig) {
+                    log.warn("Kafka {}/{} has invalid spec.cruisecontrol.config: {}",
+                            kafkaAssembly.getMetadata().getNamespace(),
+                            kafkaAssembly.getMetadata().getName(),
+                            error);
+                }
+                throw new InvalidResourceException("Kafka " +
+                        kafkaAssembly.getMetadata().getNamespace() + "/" + kafkaAssembly.getMetadata().getName() +
+                        " has invalid spec.cruisecontrol.config: " +
+                        String.join(", ", errorsInConfig));
+            }*/
+            //cruiseControl.setConfiguration(configuration);
+
+            cruiseControl.setUserAffinity(affinity(spec));
+            cruiseControl.setResources(spec.getResources());
+            cruiseControl.setTolerations(tolerations(spec));
             cruiseControl.setOwnerReference(kafkaAssembly);
+
+            if (spec.getTemplate() != null) {
+                CruiseControlTemplate template = spec.getTemplate();
+
+                ModelUtils.parsePodTemplate(cruiseControl, template.getPod());
+
+                if (template.getCruiseControlContainer() != null && template.getCruiseControlContainer().getEnv() != null) {
+                    cruiseControl.templateCruiseControlContainerEnvVars = template.getCruiseControlContainer().getEnv();
+                }
+
+                if (template.getTlsSidecarContainer() != null && template.getTlsSidecarContainer().getEnv() != null) {
+                    cruiseControl.templateTlsSidecarContainerEnvVars = template.getTlsSidecarContainer().getEnv();
+                }
+
+                ModelUtils.parsePodDisruptionBudgetTemplate(cruiseControl, template.getPodDisruptionBudget());
+            }
         } else {
             cruiseControl.isDeployed = false;
         }
 
         return cruiseControl;
     }
 
-    public static StatefulSet updateKafkaConfig(Kafka kafkaAssembly, StatefulSet kafkaSs) {
+    static List<Toleration> tolerations(CruiseControlSpec spec) {
+        if (spec.getTemplate() != null
+                && spec.getTemplate().getPod() != null
+                && spec.getTemplate().getPod().getTolerations() != null) {
+            return spec.getTemplate().getPod().getTolerations();
+        }
+        return null;
+    }
+
+    static Affinity affinity(CruiseControlSpec spec) {
+        if (spec.getTemplate() != null
+                && spec.getTemplate().getPod() != null
+                && spec.getTemplate().getPod().getAffinity() != null) {
+            return spec.getTemplate().getPod().getAffinity();
+        }
+        return null;
+    }
 
+    public static StatefulSet updateKafkaConfig(Kafka kafkaAssembly, StatefulSet kafkaSs) {
```

Comment:
stuff from `Kafka`, we don't have a StatefulSet here, CC will be a Deployment, or?

Reply:
So AFAIK we can get CC store everything in Kafka Topics? But you can configure a different state store if you want. Do we want to support this or just use Kafka?

Reply:
@ppatierno CC will just be a deployment, this function is called by the KafkaAssemblyOperator when checking if the Kafka configuration section has changed [1]. Since this function checks whether the CruiseControlMetricReporter should be added or removed from the `metric.reporters` section of the Kafka configuration, I thought it would be best to put the logic in the CruiseControl class.

@tomncooper from what I understand all CC state is stored in Kafka Topics by default, so I think we should stick with that at least for now.

[1] https://github.com/kyguy/strimzi-kafka-operator/blob/a7a09f80f8d1033c82bdb65ad26bed8acc2826f1/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java#L1873-L1888